src/Model/Repository/ContactRepository.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Friendica\Model\Repository;
+
+use Friendica\Model\Entity\Contact;
+use Friendica\Model\Repository\Traits\UniqueIdSelectableTrait;
+
+class ContactRepository extends BaseRepository
+{
+	use UniqueIdSelectableTrait;
+
+	static $table = 'contact';
+	/** @var Contact */
+	static $entity = Contact::class;
+
+	/**
+	 * @param array $conditions
+	 *
+	 * @return Contact[]
+	 * @throws \Exception
+	 */
+	public function select(array $conditions)
+	{
+		return parent::select($conditions);
```

Comment:
This is what would happen if the method wasn't overloaded, what is the point? Setting the correct return type?

Reply:
Yep.. PHP doesn't support type-overloading. So the base method returns "mixed" and the specific repo has to override it. I didn't find an alternative except explicit creating these methods in each repository again and again.